### PR TITLE
Refactor of position in the carrot app

### PR DESCRIPTION
**Pull request by Buckram123**
This PR aims to refactor handling of the position inside the carrot app, to make it easier to maintain.

### CODE REVIEW

File: contracts/carrot-app/src/handlers/query.rs [hunk 1]
```diff
@@ -89,26 +85,33 @@ fn query_compound_status(deps: Deps, env: Env, app: &App) -> AppResult<CompoundS
...
     };
-    let pool = get_osmosis_position(deps, position.position_id)?;
 
-    let balances = try_proto_to_cosmwasm_coins(vec![pool.asset0.unwrap(), pool.asset1.unwrap()])?;
-    let liquidity = pool.position.unwrap().liquidity.replace('.', "");
+    let balances = try_proto_to_cosmwasm_coins(vec![
+        carrot_position.position.asset0.unwrap(),
+        carrot_position.position.asset1.unwrap(),
+    ])?;
+    // TODO: Use Decimal256 instead?
```
Comment (CyberHoward) [inline contracts/carrot-app/src/handlers/query.rs:108]: Nah, FE needs to provide as integer as well. Could use `Uint256` but serialization will be the same (String).

File: contracts/carrot-app/src/handlers/migrate.rs [hunk 2]
```diff
@@ -27,9 +28,16 @@ pub struct OldPoolConfig {
     pub asset1: AssetEntry,
 }
 
+#[cw_serde]
+pub struct OldPosition {
```
Comment (CyberHoward) [inline contracts/carrot-app/src/handlers/migrate.rs:32]: We should version this instead of calling it old. 

I.e. `V0_1Position`

Comment (CyberHoward) [reply contracts/carrot-app/src/handlers/migrate.rs]: Can we use the actual app version here? so think that's 0.3 now?

Comment (Buckram123) [reply contracts/carrot-app/src/handlers/migrate.rs]: > Can we use the actual app version here? so think that's 0.3 now? @Buckram123

0.1 to 0.3 have the same position structure, so I went with version it got introduced in

Comment (Buckram123) [reply contracts/carrot-app/src/handlers/migrate.rs]: Also keeping it this way for 0.1->1.0 migration

Comment (CyberHoward) [reply contracts/carrot-app/src/handlers/migrate.rs]: Ok good :)

**Review by CyberHoward — APPROVED** [review]

**Pull request merged**